File: verify/handmade/.proto/protoReader.cs
// 以下是模板驗證用程式碼

using System.IO;
using System.Collections.Generic;
using Google.Protobuf.Reflection; // 這為了通過編譯的程式碼, 不可使用
using Google.Protobuf; // 這為了通過編譯的程式碼, 不可使用
using pb = global::Google.Protobuf; // 這為了通過編譯的程式碼, 不可使用

namespace SheeterProto {
    public partial class RewardReader {
        public static string FileName() {
            return "reward.pbd";
        }

        public bool FromPath(string path) {
            return FromData(File.ReadAllBytes(Path.Combine(path, FileName())));
        }

        public bool FromData(byte[] data) {
            Datas = RewardStorer.Parser.ParseFrom(data);
            return Datas != null;
        }

        public IDictionary<long, Reward> Data {
            get {
                return Datas.Datas;
            }
        }

        private RewardStorer Datas = null;
    }
}

// 以下是為了通過編譯的程式碼, 不可使用

namespace SheeterProto {
    public sealed partial class Reward : pb::IMessage<Reward>
    {
        public MessageDescriptor Descriptor => throw new System.NotImplementedException();

        public int CalculateSize()
        {
            throw new System.NotImplementedException();
        }

        public Reward Clone()
        {
            throw new System.NotImplementedException();
        }

        public bool Equals(Reward? other)
        {
            throw new System.NotImplementedException();
        }

        public void MergeFrom(Reward message)
        {
            throw new System.NotImplementedException();
        }

        public void MergeFrom(CodedInputStream input)
        {
            throw new System.NotImplementedException();
        }

        public void WriteTo(CodedOutputStream output)
        {
            throw new System.NotImplementedException();
        }
    }

    public sealed partial class RewardStorer : pb::IMessage<RewardStorer>
    {
        private static readonly pb::MessageParser<RewardStorer> _parser = new pb::MessageParser<RewardStorer>(() => new RewardStorer());
        public static pb::MessageParser<RewardStorer> Parser { get { return _parser; } }
        public Dictionary<long, Reward> Datas = new Dictionary<long, Reward>();
        public MessageDescriptor Descriptor => throw new System.NotImplementedException();

        public int CalculateSize()
        {
            throw new System.NotImplementedException();
        }

        public RewardStorer Clone()
        {
            throw new System.NotImplementedException();
        }

        public bool Equals(RewardStorer? other)
        {
            throw new System.NotImplementedException();
        }

        public void MergeFrom(RewardStorer message)
        {
            throw new System.NotImplementedException();
        }

        public void MergeFrom(CodedInputStream input)
        {
            throw new System.NotImplementedException();
        }

        public void WriteTo(CodedOutputStream output)
        {
            throw new System.NotImplementedException();
        }
    }
}

File: verify/handmade/.proto/protoReader.go
// 以下是模板驗證用程式碼

package _proto

import (
	"fmt"
	"os"
	"path/filepath"

	"google.golang.org/protobuf/proto"
	"google.golang.org/protobuf/reflect/protoreflect"
)

type RewardReader struct {
	RewardStorer
}

func (this *RewardReader) FileName() string {
	return "reward.pbd"
}

func (this *RewardReader) FromPath(path string) error {
	data, err := os.ReadFile(filepath.Join(path, this.FileName()))

	if err != nil {
		return fmt.Errorf("RewardReader: from path failed: %w", err)
	}

	return this.FromData(data)
}

func (this *RewardReader) FromData(data []byte) error {
	if err := proto.Unmarshal(data, &this.RewardStorer); err != nil {
		return fmt.Errorf("RewardReader: from data failed: %w", err)
	}

	return nil
}

// 以下是為了通過編譯的程式碼, 不可使用

type RewardStorer struct {
}

func (x *RewardStorer) Reset() {
}

func (x *RewardStorer) String() string {
	return ""
}

func (*RewardStorer) ProtoMessage() {}

func (x *RewardStorer) ProtoReflect() protoreflect.Message {
	return nil
}

func (*RewardStorer) Descriptor() ([]byte, []int) {
	return nil, nil
}

func (x *RewardStorer) GetDatas() map[int64]*int {
	return nil
}
